chalice/chalicelib/matrix_handler.py

```diff
@@ -0,0 +1,165 @@
+        :param temp_dir: A temporary directory for storing all downloaded matrix files.
+        :return: A list of downloaded matrix files paths.
+        """
+        # Filter and download only matrices file that satisfies a specific suffix within bundles
+        for bundle_uuid in bundle_uuids:
+            dest_name = os.path.join(temp_dir, bundle_uuid)
+
+            try:
+                logger.info("Downloading matrices from bundle, {}, into {}."
+                            .format(bundle_uuid, dest_name))
+
+                hca_client.download(
+                    bundle_uuid=bundle_uuid,
+                    replica="aws",
+                    dest_name=dest_name,
+                    metadata_files=(),
+                    data_files=("*{}".format(self._suffix),)
+                )
+            except Exception as e:
+                raise e
+
+        # Get all downloaded mtx paths from temp_dir
+        local_mtx_paths = get_mtx_paths(temp_dir, self._suffix)
+
+        logger.info("Done downloading %d matrix files.", len(local_mtx_paths))
+
+        return local_mtx_paths
+
+    @abstractmethod
+    def _concat_mtx(self, mtx_paths: List[str], out_file: str) -> None:
+        """
+        Concatenate a list of matrices, and save into a new file on disk.
+        :param mtx_paths: A list of downloaded local matrix files paths.
+        :param out_file: Path to the concatenated matrix.
+        """
+
+    def _upload_mtx(self, path: str, request_id: str):
+        """
+        Upload a matrix file into an s3 bucket.
+        :param path: Path of the merged matrix.
+        :param request_id: Merge request id.
+        """
+        logger.info("%s", "Uploading \"{}\" to s3 bucket: \"{}\".".format(
+            os.path.basename(path),
+            MERGED_MTX_BUCKET_NAME
+        ))
+
+        with open(path, "rb") as merged_matrix:
+            try:
+                s3_blob_store.upload_file_handle(
+                    bucket=MERGED_MTX_BUCKET_NAME,
+                    key=request_id + self._suffix,
+                    src_file_handle=merged_matrix
+                )
+            except Exception as e:
+                raise e
+
+        logger.info("Done uploading.")
+
+    def run_merge_request(self, bundle_uuids: List[str], request_id: str, job_id: str) -> None:
+        """
+        Merge matrices within bundles, and upload the merged matrix to an s3 bucket.
+
+        :param bundle_uuids: Bundles' uuid for locating bundles in DSS.
+        :param request_id: Merge request id.
+        :param job_id: Job id of the request.
+        """
+        try:
+            logger.info("Concatenate matrices for request_id: {}".format(request_id))
+
+            # Print out the current usage of /tmp/ directory
+            call(["df", "-H", TEMP_DIR])
+
+            logger.info("tmp directory contains: {}".format(str(os.listdir(TEMP_DIR))))
+
+            # Clean /tmp folder before each run
+            call('rm -rf /tmp/*', shell=True)
```

Comment:
You should use `TEMP_DIR` here.

Reply:
Thanks for your reminding. I will fix it.